core/sign_in_test.go

```diff
@@ -0,0 +1,116 @@
+package core
+
+import (
+	"testing"
+)
+
+func TestGenerateSignInUriShouldGenerateCorrectUri(t *testing.T) {
+	expectedSignInUri := "https://example.com/authorize?client_id=clientId&code_challenge=codeChallenge&code_challenge_method=S256&prompt=consent&redirect_uri=https://example.com/callback&resource=resourceA&resource=resourceB&response_type=code&scope=openid offline_access profile&state=state"
+
+	signInUri, err := GenerateSignInUri(&SignInUriGenerationOptions{
+		AuthorizationEndpoint: "https://example.com/authorize",
+		ClientId:              "clientId",
+		RedirectUri:           "https://example.com/callback",
+		CodeChallenge:         "codeChallenge",
+		State:                 "state",
+		Scopes:                []string{"openid", "offline_access", "profile"},
+		Resources:             []string{"resourceA", "resourceB"},
+		Prompt:                "consent",
+	})
+
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Noob question: why is this needed?

Reply:
We need the test to fail right now when `GenerateSignInUri` returns an error.